src/extensions/MarketCreation.sol

```diff
@@ -0,0 +1,34 @@
+pragma solidity ^0.4.13;
+
+
+// FIXME: Remove this stub once the contract is implemented in Solidity
+contract Market {
+    function stub();
+}
+
+
+// FIXME: Remove this stub once the contract is implemented in Solidity
+contract ReportingWindow {
+    function createNewMarket(int256 _endTime, int256 _numOutcomes, int256 _payoutDenominator, int256 _feePerEthInWei, address _denominationToken, address _sender, int256 _minDisplayPrice, int256 _maxDisplayPrice, address _automatedReporterAddress, int256 _topic) payable returns (Market);
+}
+
+
+// FIXME: Remove this stub once the contract is implemented in Solidity
+contract Branch {
+    function getReportingWindowByMarketEndTime(int256 _endTime, int256 _hasAutomatedReporter) returns (ReportingWindow);
+}
+
+
+contract MarketCreation {
+    function createScalarMarket(Branch _branch, int256 _endTime, int256 _feePerEthInWei, address _denominationToken, int256 _minDisplayPrice, int256 _maxDisplayPrice, address _automatedReporterAddress, int256 _topic) payable public returns (Market _market) {
+        ReportingWindow _reportingWindow = _branch.getReportingWindowByMarketEndTime(_endTime, _automatedReporterAddress != 0 ? 1 : 0);
```

Comment:
> because editors right now lack strong inference support, unlike Kotlin/C# it isn't trivial to figure out what the inferred type of something is so when it is non-obvious from context

This is actually why I would prefer not using var. As a reader of code this makes var on obstruction to immediate understanding.

I get the argument with regards to picking constrained types though so I'm not totally opposed. We should definitely come up with a rule on usage (or avoidance).